PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightsScalaProjectWizard.java

```diff
@@ -45,12 +47,11 @@
 import com.microsoft.azuretools.hdinsight.Activator;
 
 public class HDInsightsScalaProjectWizard extends JavaProjectWizard implements IExecutableExtension {
-	private SparkVersion sparkVersion;
+	private Optional<SparkVersion> sparkVersion = Optional.empty();
```

Comment:
Thanks for trying Java 8 `Optional`!

It might be better to put the raw type class `SparkVersion` here, and wrap `Optional` in the `getSparkVersion()` return value. (IJ will warn here that the `Optional` type is not serializable.)

Reply:
Thanks! Updated

---
In reply to: [153124608](https://github.com/Microsoft/azure-tools-for-java/pull/1262#discussion_r153124608) [](ancestors = 153124608)